core/src/trezor/workflow.py

```diff
@@ -15,20 +15,30 @@
 
     import micropython
 
-    from trezor import utils
-
-
-ALLOW_WHILE_LOCKED = (
-    MessageType.Initialize,
-    MessageType.EndSession,
-    MessageType.GetFeatures,
-    MessageType.Cancel,
-    MessageType.LockDevice,
-    MessageType.DoPreauthorized,
-    MessageType.WipeDevice,
-    MessageType.SetBusy,
-    MessageType.Ping,
-)
+if utils.USE_THP:
+    ALLOW_WHILE_LOCKED = (
+        MessageType.ThpCreateNewSession,
```

Comment:
This breaks `test_autolock_interrupts_passphrase` - manually: `trezorctl set auto-lock-delay 10` then create a new session but don't finish typing the password, the keyboard stays unlocked.

I wonder if we can just remove the line? Seedless sessions don't go through the handler and it calls `unlock_device()` near the beginning anyway?